src/index.ts

```diff
@@ -0,0 +1,255 @@
+import { WebSocket, WebSocketServer } from 'ws';
+import { IRegRequest, IRegResponce, IServerMessage } from './interfaces';
+import { Game } from './game';
+import { User } from './user';
+import { Player } from './player';
+
+const wss = new WebSocketServer({ port: 3000 });
+const messages: Array<string> = [];
+const games: Array<Game> = [];
+const users: Array<User> = [];
+
+const sendGames = (ws: WebSocket) => {
+  const result = {
+    id: 0,
+    type: "update_room",
+    data: JSON.stringify(
+      games.map((_game, gameIndex) => {
+        return {
+          roomId: gameIndex,
+          roomUsers: _game.players.map((player, playerIndex) => ({
+            name: player.user.name,
+            index: playerIndex
+          }))
+        }
+      }))
+  };
+  console.log(result);
+
+  ws.send(JSON.stringify(result
+  ))
+}
+
+const handlers = {
+  reg: async (_data: string, ws: WebSocket) => {
+    const data: IRegRequest = JSON.parse(_data);
+    console.log(data);
+
+    const userRegIndex = users.findIndex(it => it.name == data.name);
+    const userReg = users[userRegIndex];
+    if (!userReg) {
+      const user = new User(data.name, data.password);
+      user.updateConnection(ws);
+      users.push(user);
+      const result: IRegResponce = {
+        name: data.name,
+        index: users.length - 1,
+        error: false,
+        errorText: '',
+      }
+      sendGames(ws);
+      return result;
+    } else if (userReg.password == data.password) {
+      userReg.updateConnection(ws);
+      const result: IRegResponce = {
+        name: data.name,
+        index: userRegIndex,
+        error: false,
+        errorText: '',
+      }
+      sendGames(ws);
+      return result;
+    } else {
+      const result: IRegResponce = {
+        name: data.name,
+        index: -1,
+        error: true,
+        errorText: 'Invalid password',
+      }
+      return result;
+    }
+  },
+
+  create_room: async (data: '', ws: WebSocket) => {
+    const userIndex = users.findIndex(it => it.connection == ws);
+    const user = users[userIndex];
+    if (!user) return;
+
+    const game = new Game();
+    game.players.push(new Player(user));
+    games.push(game);
+    wss.clients.forEach(it => {
```

Comment:
Rooms list should be sent to logged in users only (except the one who created that room).